src/UiModal.vue

```diff
@@ -176,6 +179,7 @@ export default {
             if (!this.dismissible) {
                 return;
             }
+            if(this.beforeClose && !this.beforeClose(this)) return;
```

Comment:
Let's make it an explicit check for `false` so you don't have to return anything if you want the close to proceed.

```suggestion
            if (this.beforeClose && this.beforeClose(this) === false) {
                return;
            }
```